implementations/ruby-json_schemer/bowtie_json_schemer.rb

```diff
@@ -56,23 +70,36 @@ class UnsupportedDialect < StandardError; end
     }
   when 'dialect'
     dialect = request.fetch('dialect')
-    @meta_schema = JSONSchemer::META_SCHEMAS_BY_BASE_URI_STR[dialect]
-    raise UnsupportedDialect, dialect unless @meta_schema
+
+    if Gem::Version.new('2.0.0') <= JSON_SCHEMER_VERSION
```

Comment:
Same here, I'd try to move this logic above to one area (so define some callable there depending on version which gives you what you need.